Source/WebKit/Platform/IPC/StreamConnectionEncoder.h

```diff
@@ -39,8 +39,9 @@ template<typename, typename> struct ArgumentCoder;
 //
 class StreamConnectionEncoder final {
 public:
-    // Stream message needs to be at least size of StreamSetDestinationID message.
-    static constexpr size_t minimumMessageSize = sizeof(MessageName) + sizeof(uint64_t);
+    // Stream allocation needs to be at least size of StreamSetDestinationID message at any offset % messageAlignment.
+    // StreamSetDestinationID has MessageName+uint64_t, where uint64_t is expected to to be aligned at 8.
+    static constexpr size_t minimumMessageSize = 16;
```

Comment:
Is it possible that we need 17?

MessageName is 2-byte aligned, so we could end up needing 3-bytes to write it. That puts us on a guaranteed 2-byte boundary, so we could need to align another 6, before writing the 8 of uint64_t.

I'd probably prefer this to be written as `(alignof(MessageName) - 1) + sizeof(MessageName) + (alignof(uint64_t) -1) + sizeof(uint64_t)`.

As discussed on Slack, it'd also be nice if `uint64_t` were replaced with something strongly typed, but that's probably out of scope for this PR.

Reply:
>Is it possible that we need 17. MessageName is 2-byte aligned, so we could end up needing 3-bytes to write it. 

No, the message allocation itself is currently aligned per messageAlignment, see the constant one line below.

>I'd probably prefer this to be written as (alignof(MessageName) - 1) + sizeof(MessageName) + (alignof(uint64_t) -1) + sizeof(uint64_t).

It's a valid idea, but I thought it was a source of bugs in its own. Your wording about where MessageName ends and possible alignment values for uint64_t was exactly correct, but your formula was not.

When done correctly, the reader needs to think as hard or harder at the formula as they need for the value 16. 

As an anecdotal note, you wrote `1+2+7+8` which is 18 when you thought you wrote 17. :)

But that's a wrong formula, even accounting for the above information. So it's not as easy to get right or to read as you'd think.  `sizeof(MessageName) + (alignof(uint64_t)-1) + sizeof(uint64_t)`  is `2 + 7 + 8 == 17`.

So if we'd simplify and accept that MessageName is always aligned, the generic idea of the formula would be that the uint64_t align factor depends on the message name size mod uint64_t alignment. So something like:
`(sizeof(MessageName) + (alignof(uint64_t) - (sizeof(MessageName) % alignof(uint64_t)) + sizeof(uint64_t)`

Which quite frankly starts to get quite complex for what it accomplishes.